R/chart.TimeSeries.R

```diff
@@ -146,169 +146,139 @@
-          period.areas = NULL, 
-          event.color = "darkgray", 
-          period.color = "aliceblue", colorset = (1:12), 
-          pch = (1:12), 
-          legend.loc = NULL, 
-          ylog = FALSE, 
-          cex.axis=0.8, 
-          cex.legend = 0.8, 
-          cex.lab = 1, 
-          cex.labels = 0.8, 
-          cex.main = 1, 
-          major.ticks='auto', 
-          minor.ticks=TRUE, 
-          grid.color="lightgray", 
-          grid.lty="dotted", 
-          xaxis.labels = NULL,
-          dygraphPlot=FALSE,
-          yaxis.pct=FALSE)
-{ # @author Peter Carl, Brian Peterson
-
-    # DESCRIPTION:
-    # Draws a line chart and labels the x-axis with the appropriate dates.
-    # This is really a "primitive", since it constructs the elements of a plot
-    # to provide lines for each column of data provided.  
-
-    # Inputs:
-    # R = assumes that data is a regular time series, not irregular.  Can take
-    # any type of object, whether a matrix, data frame, or timeSeries.
-    # date.format: allows passing of a date format for the xaxis
-    # legend.loc = use this to locate the legend, e.g., "topright"
-    # colorset = use the name of any of the palattes above
-    # reference.grid = if true, draws a grid aligned with the points on the
-    #    x and y axes.
-    # xaxis = if true, draws the x axis.
-    # event.lines = if not null, will draw vertical lines indicating that an
-    #    event happened during that time period.  event.lines should be a list
-    #    of dates (e.g., c("09/03","05/06")) formatted the same as date.format.
-    #    This function matches the re-formatted row names (dates) with the
-    #    events.list, so to get a match the formatting needs to be correct.
-    # event.labels = if not null and event.lines is not null, this will apply
-    #    labels to the vertical lines drawn.
-
-    # All other inputs are the same as "plot" and are principally included
-    # so that some sensible defaults could be set.
-
-    # Output:
-    # Draws a timeseries graph of type "line" with some sensible defaults.
-
-    # FUNCTION:
-
-    y = checkData(R)
+#' 
 
-    # Set up dimensions and labels
-    columns = ncol(y)
-    rows = nrow(y)
-    columnnames = colnames(y)
 
-    if (is.null(date.format)){
-	freq = periodicity(y)
-	yr_eq <- ifelse(format(index(first(y)),format="%Y")==format(index(last(y)),format="%Y"),TRUE,FALSE) 
-	switch(freq$scale,
-	    seconds = { date.format = "%H:%M"},
-	    minute = { date.format = "%H:%M"},
-	    hourly = {date.format = "%d %H"},
-	    daily = {if (yr_eq) date.format = "%b %d" else date.format = "%Y-%m-%d"},
-	    weekly = {if (yr_eq) date.format = "%b %d" else date.format = "%Y-%m-%d"},
-	    monthly = {if (yr_eq) date.format = "%b" else date.format = "%b %y"},
-	    quarterly = {if (yr_eq) date.format = "%b" else date.format = "%b %y"},
-	    yearly = {date.format = "%Y"}
-	)
+# New Function with multi-engine option
+chart.TimeSeries <-
```

Comment:
Running this fails for me: chart.TimeSeries(edhec, plot_engine = "plotly")

With the following error:
'Error in layout(., title = main) : unused argument (title = main)'

Reply:
Updates are made for pass by dots, does the issue still exist?